Automattic/wp-calypso
# BackupBrowser: Add granular restore page skeleton

## Proposed Changes

* Setup new `/backup/:site/granular-restore/:rewindId` route
* Add a new `BackupGranularRestoreFlow` component that renders the granular restore confirmation page skeleton
* `Go back` and `Cancel` button redirect back the customer to the backup browser page. 
* Add `Restore` button in the backup browser pointing to the new route.
* `Confirm restore` button just `console.log` the current selected files

## Screenshot
| Component | Screenshot |
|---|---|
| Backup Browser restore button | ![CleanShot 2023-08-31 at 22 40 06@2x](https://github.com/Automattic/wp-calypso/assets/1488641/9a555b0f-4835-40c5-a23e-33de2c71a54b) |
| Granular restore confirmation page | <img width="713" alt="CleanShot 2023-08-31 at 23 15 09@2x" src="https://github.com/Automattic/wp-calypso/assets/1488641/8c50c4c6-5ad6-412a-a041-3b2660bd1510"> |

## Testing Instructions
* Start a Jetpack Cloud or Calypso Live branch and add `?flags=jetpack/backup-granular` to the URL after visiting the Backup Browser.
* Pick a file and ensure the `Restore` button is visible (except on the WordPress item).
* Click on `Restore` and it should redirect you to `/backup/:site/granular-restore/:rewindId`.
* Ensure the file you selected is displayed in the granular restore page.
* Test `Go Back` and `Cancel` button. They should redirect you back to the backup browser for the selected rewind ID.
* Click on `Confirm restore` and check your developer console. It should log the include/exclude lists and only the file selected should be there.

## Pre-merge Checklist
